Dockerfiles/agent/cont-init.d/50-ecs-managed.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Quick exit unless ECS managed instances are explicitly enabled
+if [[ "${ECS_MANAGED_INSTANCES:-}" != "true" ]]; then
+    exit 0
+fi
+
+# Require DD_ECS_DEPLOY_MODE=sidecar
+if [[ "${DD_ECS_DEPLOYMENT_MODE:-}" != "sidecar" ]]; then
+  exit 0
+fi
+
+# Set a default config for ECS Managed Instances if found
+# Don't override /etc/datadog-agent/datadog.yaml if it exists
+if [[ ! -e /etc/datadog-agent/datadog.yaml ]]; then
+    ln -s  /etc/datadog-agent/datadog-ecs.yaml \
+           /etc/datadog-agent/datadog.yaml
+fi
+
+# Remove all default checks
+find /etc/datadog-agent/conf.d/ -iname "*.yaml.default" | xargs grep -L 'ad_identifiers' | xargs rm -f
```

Comment:
Potential issues:
- due to the double-quotes, `*.yaml.default` glob pattern may be evaluated too early, i.e. by `bash` instead of `find`,
- if any filename contains spaces, newlines, or special characters, the pipeline breaks,
- if `find` returns no results, `xargs grep` may run with no arguments,
- if `grep -L` finds no matches, `xargs rm` may run without files.

Suggestions:
```suggestion
find /etc/datadog-agent/conf.d -iname '*.yaml.default' -print0 | \
    xargs -0 -r grep -LZ ad_identifiers | \
    xargs -0 -r rm -f
```
or, simply: (no pipeline)
```suggestion
find /etc/datadog-agent/conf.d -iname '*.yaml.default' -exec grep -L ad_identifiers {} + -delete
```

Reply:
https://github.com/DataDog/datadog-agent/pull/42203/commits/4d92d47ba6c7050c8d8bb3f9e800e9923c511f6a
I opted for the first suggestion as the second one could potentially remove files with ad_identifiers.